adapter.go

```diff
@@ -611,6 +613,33 @@ func (a *Adapter) AddPolicies(sec string, ptype string, rules [][]string) error
 	return a.db.Create(&lines).Error
 }
 
+// Transaction perform a set of operations within a transaction
+func (a *Adapter) Transaction(e casbin.IEnforcer, fc func(casbin.IEnforcer) error, opts ...*sql.TxOptions) error {
+	var err error
+	oriAdapter := a.db
+	// reload policy from database to sync with the transaction
+	defer func() {
+		e.SetAdapter(&Adapter{db: oriAdapter})
+		err = e.LoadPolicy()
+		if err != nil {
+			panic(err)
+		}
+	}()
+	copyDB := *a.db
+	tx := copyDB.Begin(opts...)
+	b := &Adapter{db: tx}
+	// copy enforcer to set the new adapter with transaction tx
+	copyEnforcer := e
```

Comment:
I'm curious here that this `e` seems to be a pointer and Won't any changes to `copyEnforcer` be applied to the origin `e`?

Reply:
Yes, you're correct. Therefore, I reset e's adapter in the defer function. The use of the copyEnforcer just makes the code more readable.

Reply:
Got it.Thanks for reply!